packages/block-editor/src/store/reducer.js

```diff
@@ -772,7 +799,8 @@ export const blocks = pipe(
 	withBlockReset,
 	withPersistentBlockChange,
 	withIgnoredBlockChange,
-	withResetControlledBlocks
+	withResetControlledBlocks,
+	withPatternCategory
```

Comment:
It seems like this is running before the `attributes` function below, so the below function is overwriting the attribute changes we're making.

Reply:
I think this means we can't use a separate reducer, so I built it into the attributes reducer.